app/src/middleware/authentication.js

```diff
@@ -0,0 +1,57 @@
+const jwt = require('jsonwebtoken');
+const Problem = require('api-problem');
+
+const log = require('../components/log')(module.filename);
+const config = require('config');
+
+/**
+ * @function _spkiWrapper
+ * Wraps an SPKI key with PEM header and footer
+ * @param {string} spki The PEM-encoded Simple public-key infrastructure string
+ * @returns {string} The PEM-encoded SPKI with PEM header and footer
+ */
+const _spkiWrapper = (spki) => `-----BEGIN PUBLIC KEY-----\n${spki}\n-----END PUBLIC KEY-----`;
+
+/**
+ * @function tokenValidator
+ *
+ * This middleware validates the token in the Authorization header of the request
+ * We have to be careful that this function does not throw an unhandled exception
+ * because of our 'shutdown' behaviour enforced at a global level in /app.js
+ * Consequently, we wrap this logic in a try/catch and return errors as the http response
+ *
+ * @param {object} req Express Request Object
+ * @param {object} res Express Response Object
+ * @param {function} next Express Callback Function
+ */
+async function tokenValidator(req, res, next) {
+  try {
+    const authorization = req.get('Authorization');
+    const bearerToken = authorization.substring(7);
+
+    if (config.has('keycloak.publicKey') &&
+      config.has('keycloak.serverUrl') &&
+      config.has('keycloak.realm')) {
```

Comment:
`config.has()` will return `true` even if the config contains `false`, since it only checks whether it *exists* and not what value it holds (see https://github.com/bcgov/common-object-management-service/pull/245).

You would have to use `config.get()` to actually get the value then convert that into a boolean (or you could steal `utils.isTruthy()` from COMS to do that for you).